From a0e5e050dfbdb672459f74bf52562bc8fc50c3b9 Mon Sep 17 00:00:00 2001
From: James Browning <jamesb.fe80@gmail.com>
Date: Mon, 7 Jun 2021 09:29:39 -0700
Subject: ntpleapfetch: Fix hash computation

---
diff --git a/ntpclients/ntpleapfetch b/ntpclients/ntpleapfetch
index 5e8500b92..039b8a682 100755
--- a/ntpclients/ntpleapfetch
+++ b/ntpclients/ntpleapfetch
@@ -310,7 +310,7 @@ verifySHA1() {
     # or maybe a space and a tab.  remove the tab, wherever it may be.  Add some 0x
     # so it can be run through printf to restore missing leading zeros.
     #
-    local FSHA1="`grep '^#h' $1 | tr -d '[:cntrl:]' | sed -e's/^#h/0x/' -e's/ / 0x/g'`"
+    local FSHA1="`grep '^#h' $1 | tr -d '[:cntrl:]' | sed -e's/^#h[ \t]*/0x/' -e's/ / 0x/g'`"
     FSHA1=`printf '%08x%08x%08x%08x%08x' $FSHA1`
 
     if [ -n "$FSHA1" -a \( "$FSHA1" = "$DSHA1" \) ]; then
