From 71c34527139e3283013844bd3b52493d2120663f Mon Sep 17 00:00:00 2001
From: Hal Murray <hmurray@megapathdsl.net>
Date: Wed, 25 Aug 2021 02:24:53 -0700
Subject: [PATCH 26/55] Cleanup startup logging

---
 ntpd/ntpd.c | 24 ++++++++++++++++++++----
 1 file changed, 20 insertions(+), 4 deletions(-)

diff --git a/ntpd/ntpd.c b/ntpd/ntpd.c
index 569ce9ea1..ba52357b6 100644
--- a/ntpd/ntpd.c
+++ b/ntpd/ntpd.c
@@ -502,11 +502,19 @@ main(
 # endif
 
 	init_logging(progname, NLOG_SYNCMASK, true);
+
+	if (!dumpopts) {
+		/* log to syslog before setting up log file */
+		announce_starting();
+	}
+
 	/* honor -l/--logfile option to log to a file */
 	if (logfilename != NULL) {
 		syslogit = false;
 		termlogit = false;
 		change_logfile(logfilename, false);
+		/* Repeat critical info in logfile. Helps debugging. */
+		announce_starting();
 	} else {
 		if (nofork)
 		    termlogit = true;
@@ -514,10 +522,6 @@ main(
 			syslogit = false;
 	}
 
-	if (!dumpopts) {
-		announce_starting();
-	}
-
 	uid = getuid();
 	if (uid && !dumpopts) {
 		termlogit = true;
@@ -880,6 +884,18 @@ main(
         /* unreachable, mainloop() never returns */
 }
 
+
+/* This goes to syslog.
+ * And again to a log file if you are using one.
+ *
+ * The first copy also goes to stderr.
+ * systemd adds that to syslog.
+ *
+ * Switching log files also logs a message before switching.
+ *
+ * If using a log file, there should be enough info in syslog
+ * to debug things with minimal extra clutter.
+ */
 void announce_starting() {
 	char buf[1024];	/* Secret knowledge of msyslog buf length */
 	char *cp = buf;
-- 
2.25.1

