bug #249216

Index: ntp-4.2.6.p3+dfsg/ntpdate/ntpdate.c
===================================================================
--- ntp-4.2.6.p3+dfsg.orig/ntpdate/ntpdate.c	2011-06-03 15:26:03.000000000 +0000
+++ ntp-4.2.6.p3+dfsg/ntpdate/ntpdate.c	2011-06-03 15:26:10.000000000 +0000
@@ -1742,8 +1742,7 @@
 		if (res->ai_family == AF_INET6)
 			if (setsockopt(fd[nbsock], IPPROTO_IPV6, IPV6_V6ONLY, (void*) &optval, sizeof(optval)) < 0) {
 				   msyslog(LOG_ERR, "setsockopt() IPV6_V6ONLY failed: %m");
-					exit(1);
-					/*NOTREACHED*/
+				/* exit(1); */
 		}
 #endif
 
